packages/babel-traverse/src/path/introspection.ts

```diff
@@ -385,8 +385,11 @@ export function _guessExecutionStatusRelativeToDifferentFunctions(
   this: NodePath,
   target: NodePath,
 ): RelativeExecutionStatus {
+  const targetIsGenericFunctionExpression =
+    (target.isArrowFunctionExpression() || target.isFunctionExpression()) &&
+    t.isVariableDeclarator(target.parent);
```

Comment:
I realized that if `target.parent.id` is not an identifier, we should bail out.

Text example:
```js
var { length } = function f(a, b, c) {};
```

Reply:
Is this also true if it's `const`?